### ISSUE TITLE
Binary download URL being populated with wrong architecture on Apple Silicon M1 Mac

### ISSUE THREAD

**Issue by LColman-SEEK** [chronotc/monorepo-diff-buildkite-plugin#83]
# Bug: all `$(arch)`'s _other than_ `aarch64` are _assumed_ to be `amd64`
Issue observed on MacBook Pro (16-inch, 2021) running **Apple M1 Pro** chip with macOS Monterey when running the buildkite agent locally.

This is causing the plugin to assemble and download the binary for the wrong architecture.  In my case (on `arm64`) it is getting: `monorepo-diff-buildkite-plugin-linux-amd64`.
This then results in an error similar to the following:
```shell
/var/folders/p5/2smnxkqs3fv27_5g43jg06_q0000rq/T/buildkite-plugins-906434519/github-com-chronotc-monorepo-diff-buildkite-plugin-v2-1-2/hooks/command:
line 27: ./monorepo-diff-buildkite-plugin-linux-amd64: cannot execute binary file
```
With "cannot execute binary file" here being the error that the ELF binary is for the wrong machine architecture.

## Minimum Steps to Reproduce
The following `pipeline.yml` logic is sufficient to cause the error (when run on an M1 Mac using local agent).
```yaml
steps:
  - plugins:
      - chronotc/monorepo-diff#v2.1.2:
          watch:
            - path: "/dummy-value"
              config:
                command: "arch"
```

### On Apple Silicon
Output of `arch` on my system:
```shell
arm64
```

Output of `uname -mps` on my system:
```shell
Darwin arm64 arm
```

### In the wild
According to [some answers on StackOverflow](https://stackoverflow.com/questions/45125516/possible-values-for-uname-m), as well as [on Wikipedia](https://en.wikipedia.org/wiki/Uname); there are many values for the system architecture.  Even just for the ARM and x86_64 / amd64 instruction sets.
At a minimum, I would think the `amd64`, `x86_64`, `aarch64` and `arm64` values from `arch` (or `uname -p`) should be handled correctly to ensure trouble free adoption of the plugin.

**Comment by germs12** [comment]
This is also affecting Intel Macs.

**Comment by xzyfer** [comment]
We have a fix in #84 we'll try to release this week.

**Comment by xzyfer** [comment]
Should be fixed in v2.2.1
